src/base/http/server.cpp

```diff
@@ -30,71 +30,82 @@
 
 #include "server.h"
 
+#include <algorithm>
+
 #include <QMutableListIterator>
 #include <QNetworkProxy>
+#include <QSslCipher>
+#include <QSslSocket>
 #include <QStringList>
 #include <QTimer>
-#ifndef QT_NO_OPENSSL
-#include <QSslSocket>
-#else
-#include <QTcpSocket>
-#endif
 
 #include "connection.h"
 
-static const int KEEP_ALIVE_DURATION = 7 * 1000;  // milliseconds
-static const int CONNECTIONS_LIMIT = 500;
-static const int CONNECTIONS_SCAN_INTERVAL = 2;  // seconds
+namespace
+{
+    QList<QSslCipher> safeCipherList()
+    {
+        const QStringList badCiphers = {"idea", "rc4"};
+        const QList<QSslCipher> allCiphers = QSslSocket::supportedCiphers();
+        QList<QSslCipher> safeCiphers;
+        for (const QSslCipher &cipher : allCiphers) {
+            bool isSafe = true;
+            for (const QString &badCipher : badCiphers) {
+                if (cipher.name().contains(badCipher, Qt::CaseInsensitive)) {
+                    isSafe = false;
+                    break;
+                }
+            }
+
+            if (isSafe)
+                safeCiphers += cipher;
+        }
+
+        return safeCiphers;
+    }
+
+    const int KEEP_ALIVE_DURATION = 7 * 1000;  // milliseconds
```

Comment:
Please move constants above the functions.

Reply:
Done.